src/Features/CSharp/Portable/FileHeaders/CSharpFileHeaderHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.FileHeaders;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.FileHeaders
+{
+    /// <summary>
+    /// Helper class used for working with file headers.
+    /// </summary>
+    internal sealed class CSharpFileHeaderHelper : AbstractFileHeaderHelper
+    {
+        public static CSharpFileHeaderHelper Instance { get; } = new CSharpFileHeaderHelper();
+
+        private CSharpFileHeaderHelper()
+        {
+        }
+
+        internal override int SingleLineCommentTriviaKind => (int)SyntaxKind.SingleLineCommentTrivia;
+        internal override int MultiLineCommentTriviaKind => (int)SyntaxKind.MultiLineCommentTrivia;
+        internal override int WhitespaceTriviaKind => (int)SyntaxKind.WhitespaceTrivia;
+        internal override int EndOfLineTriviaKind => (int)SyntaxKind.EndOfLineTrivia;
+        internal override string CommentPrefix => "//";
+
+        protected override string TryGetTextContextOfComment(SyntaxTrivia commentTrivia)
+        {
+            if (commentTrivia.IsKind(SyntaxKind.SingleLineCommentTrivia))
+            {
+                return commentTrivia.ToFullString().Substring(2);
+            }
+            else if (commentTrivia.IsKind(SyntaxKind.MultiLineCommentTrivia))
+            {
+                var triviaString = commentTrivia.ToFullString();
+
+                var startIndex = triviaString.IndexOf("/*", StringComparison.Ordinal) + 2;
+                var endIndex = triviaString.LastIndexOf("*/", StringComparison.Ordinal);
+                if (endIndex == -1)
+                {
+                    // While editing, it is possible to have a multiline comment trivia that does not contain the closing '*/' yet.
+                    return null;
```

Comment:
should we just take the end position then?

Reply:
what about `/*/`?

Reply:
I'll add tests covering the case. I don't think we need to change behavior though.

Reply:
I did change the behavior here, and added several new tests